Dataset.py

```diff
@@ -588,6 +623,8 @@ def _base_init(self):
 
   def initialize(self):
     """
+    TODO: Pycharm gives some weird usage places here...
```

Comment:
What is this TODO about?
(Probably you speak about cases where PyCharm cannot correctly infer the type of some object. Then it cannot know whether this is this `initialize`, or from some other class, so it shows it (under "unclassified usages" or sth like that). This is the best thing it can do. You could go through this list and check why PyCharm could not correctly infer the type. But that's other code, not related to here in `Dataset.py`. There is no problem here in `Dataset.py`.)